railties/lib/rails/application/configuration.rb

```diff
@@ -162,6 +162,10 @@ def load_defaults(target_version)
 
           self.autoloader = :zeitwerk if %w[ruby truffleruby].include?(RUBY_ENGINE)
 
+          if respond_to?(:active_support)
+            active_support.legacy_format = false
```

Comment:
This should have a corresponding entry in the [new framework defaults initializer](https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_6_1.rb.tt).

It isn't obvious what setting `active_support.legacy_format = false` will do, maybe `active_support.optimized_cache_entry_format = true` instead?

Reply:
Yeah, I wasn't too happy with it either, but wasn't inspired at all. I'll take your suggestion.